Create group "edit" function

Tai-f opened the pull request:
#What
1. create edit group view
2. to make "edit" group function by using “edit” & “update”controller


#Why
to make this site more useful by this function

Commit:
create pull request

Commit:
Configure routes

add “edit”, “update”

Commit:
Define "edit" method

define @group

Commit:
Edit edit.html.haml by form_for

change codes to apply ‘form_for’

Commit:
Define method 'update'

define ‘update’ method
configure strong parameter ‘name’

hc0208 commented on app/controllers/groups_controller.rb:
```diff
@@ -12,8 +12,23 @@ def create
   end
 
+  def edit
+    @group = Group.find(params[:id])
```
@group = Group.find(params[:id])が2つ出てくるので共通のメソッドとしてprivate以下に定義して、before_actionで呼び出すようにしましょう。

hc0208 commented on app/controllers/groups_controller.rb:
```diff
@@ -12,8 +12,23 @@ def create
+
+  def update
+    group = Group.find(params[:id])
+    group.update(update_params)
```
createの際と同じようにupdateに成功した時と失敗した時の処理をかくようにしましょう。

hc0208 commented on app/controllers/groups_controller.rb:
```diff
@@ -12,8 +12,23 @@ def create
+
+    def update_params
+      params.require(:group).permit(:name)
+    end
```
上のcreate_paramsと同じ記述なので、group_paramsとして同じものを用いましょう。

hc0208 commented on app/views/groups/edit.html.haml:
```diff
@@ -1,25 +1,22 @@
-    %input{:name => "utf8", :type => "hidden", :value => "✓"}/
-    %input{:name => "_method", :type => "hidden", :value => "patch"}/
-    %input{:name => "authenticity_token", :type => "hidden", :value => "U5ABp0/UobRQTSwsR0V40bX1w3t0rZ0CTqrVyLecD+L6sdhBRO/gVcKy7w/Rgxnc7Hf0Ts39996dal3to5fTag=="}/
+  = form_for(@group, action: :patch) do|f|
```
action: :patchは書かなくても問題ありません。

Commit:
Modify edit.html.haml

leave out "patch"action

Commit:
Change the name of  strong parameter

The name changed  “group_params”
due to use this in  other methods.

Commit:
define new method (strong parameters)

Define new method “set_group” as strong parameter due to use this in
other methods

Commit:
Define if method

the view move to index if group updated.
if not, the view did not move by render

Commit:
Define "move_to_index"

define “move_to_index” method due to use this in other methods.

Commit:
Add "end" to avoid errors
